datasets/lt/pep_declarations/crawler.py

```diff
@@ -0,0 +1,204 @@
+import random
+from time import sleep
+from typing import Optional
+
+from requests import HTTPError
+
+from zavod import Context, Entity
+from zavod import helpers as h
+from zavod.logic.pep import categorise
+
+DEKLARACIJA_ID_RANGE = range(301_730, 637_217)
+# sample 50 for dev purposes
+DEKLARACIJA_ID_RANGE = random.sample(DEKLARACIJA_ID_RANGE, 50)
+
+
+class PinregSession:
+    """object for interactacting with PINREG portal"""
+
+    def __init__(self, context: Context):
+        self.context = context
+        guest_id = random.randint(0, 9_999_999)  # random guest token
+        self._guest_token = f"c{guest_id:07d}"
+
+    def get_deklaracija_by_id(self, id: int) -> Optional[dict[any]]:
+        id_str = f"{id:06d}"
+        try:
+            return self.context.fetch_json(
+                url=f"https://pinreg.vtek.lt/external/deklaracijos/{id_str}/perziura/viesa",
+                params={"v": self._guest_token},
+                headers={
+                    "Accept": "application/json",
+                    "Referer": f"https://pinreg.vtek.lt/app/pid-perziura/{id_str}",
+                },
+                cache_days=30,
+            )
```

Comment:
I think we'll only fill up the cache without ever really getting to use it if we randomise the guest id. Any concerns about hardcoding the guest ID to some random-looking number?